core/lib/mcxtx_api.h

```diff
@@ -272,4 +272,287 @@
     };
     /** Storage for #MCXT_NUM_OPMASK_SLOTS mask registers as part of AVX-512. */
     dr_opmask_t opmask[MCXT_NUM_OPMASK_SLOTS];
-#endif /* ARM/X86 */
+#elif defined(RISCV64)
+    /* FIXME-RISCV: add rest of machine context and register aliases. */
+    /* Any changes in order here must be mirrored in arch/riscv64.asm offsets.
+     */
+    union {
+        /* FIXME-RISCV: This is hard-wired to zero so could be removed */
+        reg_t x0;   /**< The x0 register. */
+        reg_t zero; /**< The zero register. */
+    }; /**< The anonymous union of alternative names for x0/zero register. */
+    union {
+        reg_t x1;   /**< The x1 register. */
+        reg_t ra;
+    };
+    union {
+        reg_t x2;  /**< The x2 register. */
+        reg_t sp;  /**< The stack pointer register.*/
+        reg_t xsp; /**< The platform-independent name for the stack pointer register. */
+    };
+    union {
+        reg_t x3;   /**< The x3 register. */
+        reg_t gp;
```

Comment:
I assume we want a Doxygen comment for these alternates.

Reply:
Done.